Pr move land detector mavlink to airspeed validated

sfuhrer opened the pull request:
Alternative to https://github.com/PX4/Firmware/pull/13706, this time with moving the subscription of the FW land detector and mavlink to airspeed_validated.

SITL tested, eg FW mission where in mid-flight the ASPD_STALL is set to 100 to trigger an "airspeed invalid" declaration. The plane then keeps on flying in non-airspeed mode, and after landing the land detector correctly detects the landing. https://review.px4.io/plot_app?log=48c0f65a-35bc-4241-b3ea-5e4a0aef78db

dagar commented on src/modules/land_detector/FixedwingLandDetector.cpp:
```diff
@@ -83,7 +83,9 @@ bool FixedwingLandDetector::_get_landed_state()
 
-		_airspeed_filtered = 0.95f * _airspeed_filtered + 0.05f * _airspeed.true_airspeed_m_s;
+		// set _airspeed_filtered to 0 if airspeed data is invalid
+		_airspeed_filtered = PX4_ISFINITE(_airspeed_validated.true_airspeed_m_s) ? 0.95f * _airspeed_filtered + 0.05f *
```
Suggestion - a simple conditional is usually easier to read than everything packed into one line with the ternary operator.

You could also add a check for airspeed_validated validity (timeout) to handle the airspeed selector failing or even being stopped manually.

sfuhrer replied:
done.

dagar commented:
In another PR we should also add something to commander to require that the airspeed selector is alive and well for arming as FW or VTOL.

dagar commented:
@PX4/testflights could you give this a try the next time you fly FW or VTOL?

dannyfpv commented:
tested on pixhawk1 fixed wing

take off: no issues
stabilize: no issues
mission mode: no issues
landing: no issues
air speed : 7.0 m/s
log:
https://review.px4.io/plot_app?log=52a96023-e5c8-4054-9846-23d3351fbbde
https://review.px4.io/plot_app?log=0fe0f2db-8c13-4685-9834-f58d20dae80d
https://review.px4.io/plot_app?log=495eec7c-cba9-4e69-809e-7f5ba87f524e
https://review.px4.io/plot_app?log=04efade6-bfc1-4723-b5a4-313dcf43710b

sfuhrer commented:
Thanks for the testflights @dannyfpv !

@dagar I've addressed your comments, see 3rd commit (replaced ternary by conditional and set airspeed to 0 if airspeed_validated stops publishing). Please have a look at that.

sfuhrer commented:
@dagar is this now good to be merged from your side?
